packages/core/src/views/aggregate-queue.js

```diff
@@ -0,0 +1,64 @@
+import { Kysely } from 'kysely'
+import { parse as parseLink } from 'multiformats/link'
+
+import { getDialect } from '../table/utils.js'
+import {
+  DEFAULT_LIMIT,
+} from '../table/constants.js'
+import {
+  DatabaseOperationError,
+} from '../table/errors.js'
+
+export const VIEW_NAME = 'aggregate_queue'
+
+/**
+ * @param {import('../types').DialectProps} dialectOpts
+ */
+export function createAggregateQueueView (dialectOpts) {
+  const dialect = getDialect(dialectOpts)
+
+  const dbClient = new Kysely({
+    dialect
+  })
+
+  return useAggregateQueueView(dbClient)
+}
+
+/**
+ * 
+ * @param {import('kysely').Kysely<import('../sql.generated').Database>} dbClient
+ * @returns {import('../types').AggregateQueueView}
+ */
+export function useAggregateQueueView (dbClient) {
+  return {
+    select: async (options = {}) => {
+      const limit = options.limit || DEFAULT_LIMIT
+
+      let res
+      try {
+        res = await dbClient
+          .selectFrom(VIEW_NAME)
+          .selectAll()
+          .limit(limit)
+          .execute()
+      } catch (/** @type {any} */ error) {
+        return {
+          error: new DatabaseOperationError(error.message)
+        }
+      }
+
+      /** @type {import('../types').AggregateOutput[]} */
+      // @ts-expect-error sql created types for view get optional
+      // while in practise they will always have a value
+      const aggregagteQueue = res.map(aggregate => ({
+        link: aggregate.link !== null && parseLink(aggregate.link),
+        size: aggregate.size != null && Number.parseInt(aggregate.size),
```

Comment:
There is a comment above for this

```js
// @ts-expect-error sql created types for view get optional
// while in practise they will always have a value
```

SQL generate sadly adds a `NULL`. I think that it does not work nicely with JOINS

Reply:
All the API receives links and puts into string on all write side, so in practise it won't ever happen not to parse. We can consider adding an error anyway

Reply:
I need to add a bunch of ignores to do so

```js
 /** @type {import('../types').AggregateOutput[]} */
      // @ts-expect-error sql created types for view get optional
      // while in practise they will always have a value
      const aggregagteQueue = res.map(aggregate => ({
        // @ts-expect-error sql created types for view get optional
        link: parseLink(/** @type {string} */ aggregate.link),
        // @ts-expect-error sql created types for view get optional
        size: /** @type {number} */(Number.parseInt(aggregate.size)) | 0,
        inserted: aggregate.inserted,
      }))
```

Because view types generate

```
export interface AggregateQueue {
  link: string | null;
  size: Int8 | null;
  inserted: Timestamp | null;
}
```

Also `Int8` is `ColumnType<string, string | number | bigint, string | number | bigint>;` which requires the `parseInt`